cmd/profiling.go

```diff
@@ -0,0 +1,102 @@
+/*
+This file is part of REANA.
+Copyright (C) 2022 CERN.
+
+REANA is free software; you can redistribute it and/or modify it
+under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"runtime"
+	"runtime/pprof"
+
+	"github.com/spf13/pflag"
+)
+
+var (
+	profileMode   string
+	profileOutput string
+)
+
+func addProfilerFlags(flags *pflag.FlagSet) {
+	flags.StringVar(
+		&profileMode,
+		"profile",
+		"none",
+		"Enable profiling. One of (none|cpu|heap)",
+	)
+	flags.StringVar(
+		&profileOutput,
+		"profile-output",
+		"profile.pprof",
+		"Name of the file to write the profile to",
```

Comment:
Not sure if we need a separate flag to control the output file, since we print all the persistent flags in help message (`go run .`) it kind of pollutes the output. We could remove it to keep it simple

Reply:
Perhaps we could reuse the `--profile` option value?  E.g. if `--profile=cpu`, then we cloud create `profile-cpu.pprof`, and ditto for `--profile=heap`, etc.